cpp/csp/python/PyStructToJson.cpp

```diff
@@ -198,12 +198,38 @@ rapidjson::Value toJsonRecursive( const StructPtr& self, rapidjson::Document& do
     return new_dict;
 }
 
-rapidjson::Value pyDictKeyToName( PyObject * py_key, rapidjson::Document& doc )
+std::pair<const char *, Py_ssize_t> pyObjectToString( PyObject * py_obj )
 {
-    // Only support strings, ints, and floats as keys
+    auto * str_obj = PyObject_Str( py_obj );
```

Comment:
`PyObject_Str` returns a new reference so I believe this leaks, we need to decref at the end of the function (or better yet use a `PyObjectPtr`).